File: Array/27-Remove-Element.go
package main

import "fmt"

func removeElement(nums []int, val int) int {

	resul := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			nums[resul] = nums[i]
			resul++
		}
	}
	return resul
}

func main() {
	val := 2
	nums := []int{0, 1, 2, 2, 3, 0, 4, 2}

	result := removeElement(nums, val)

	fmt.Print("Масссив: ")

	for i := 0; i < result; i++ {
		fmt.Print(nums[i])
	}
}

File: Array/27-Remove Element.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {

        int result = 0;
        for (int i = 0; i < nums.size(); ++i){
            if (nums[i] != val){
                nums[result] = nums[i];
                cout << "result: " << result << endl;
                result++;
            }
        }
        return result;
    }
};

int main(){
    int val = 2;
    vector<int> array { 0,1,2,2,3,0,4,2 };

    Solution sol;

    int resul = sol.removeElement(array, val);
    
    cout << "============================" << endl;

    cout << resul << endl;

    for (int i = 0; i < resul; ++i){
        cout << array[i] << endl;
    }

    return 0;
}